src/main/java/model/ModelGraph.java

```diff
@@ -130,12 +126,12 @@ public void deleteEdge(String edgeId){
         return Optional.ofNullable(edges.get(id));
     }
 
-    public Optional<Vertex> getVertexBetween(Vertex beginning, Vertex end) {
+    public List<Vertex> getVertexBetween(Vertex beginning, Vertex end) {
```

Comment:
Here we have a problem with naming convenction. If we expect more than one vertex between given params and we use List, then method name should be getVertexes. On the other hand I think using of Optionals is more consistent with the rest of our code and may someone else already has used this method.